dali/imgcodec/decoders/libtiff/tiff_libtiff.cc

```diff
@@ -304,23 +330,52 @@ DecodeResult LibTiffDecoderInstance::Decode(DecodeContext ctx,
       using InputType = detail::depth2type<InTypeBits>::type;
 
       kernels::DynamicScratchpad scratchpad;
-      InputType *row_in;
+      InputType *in;
       if (info.bit_depth == InTypeBits) {
-        row_in = static_cast<InputType *>(row_buf.get());
+        in = static_cast<InputType *>(buf.get());
       } else {
-        row_in = static_cast<InputType*>(scratchpad.Alloc(mm::memory_kind_id::host,
-                   volume(in_line_shape) * sizeof(InputType), sizeof(InputType)));
+        in = static_cast<InputType*>(scratchpad.Alloc(mm::memory_kind_id::host,
+                  volume(tile_shape) * sizeof(InputType), sizeof(InputType)));
       }
 
       OutType *const img_out = out.mutable_data<OutType>();
-      for (int64_t roi_y = 0; roi_y < roi.shape()[0]; roi_y++) {
-        LIBTIFF_CALL(TIFFReadScanline(tiff.get(), row_buf.get(), roi.begin[0] + roi_y, 0));
-        if (info.bit_depth != InTypeBits) {
-          detail::UnpackBits(info.bit_depth, row_in, row_buf.get(), volume(in_line_shape));
+
+      int64_t first_tile_y = roi.begin[0] - roi.begin[0] % info.tile_height;
+      int64_t first_tile_x = roi.begin[1] - roi.begin[1] % info.tile_width;
```

Comment:
I would add a comment, that for non-tiled we always start from 0, as we read whole lines.

Reply:
Added